File: android/app/src/main/kotlin/com/typography/animations/loop/Ripple.kt
package com.typography.animations.loop

import com.typography.AnimatedTextState

fun evaluateRipple(timeMs: Long, durationMs: Int): AnimatedTextState {
    val phase = (timeMs % durationMs) / durationMs.toFloat()
    return AnimatedTextState(typewriterProgress = phase)
}

File: lib/utils/animations/loop/ripple.dart
import '../base/animated_text_state.dart';

AnimatedTextState evaluateRipple(int timeMs, int durationMs) {
  final safeD = durationMs > 0 ? durationMs : 100;
  final phase = (timeMs % safeD) / safeD.toDouble();
  return AnimatedTextState(typewriterProgress: phase);
}
